src/OrbitBase/include/OrbitBase/StopState.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_STOP_STATE_H_
+#define ORBIT_BASE_STOP_STATE_H_
+
+#include <absl/synchronization/mutex.h>
+
+namespace orbit_base_internal {
+
+// StopState is an implementation detail of StopSource and StopToken and is not intended to be used
+// outside of StopSource and StopToken
+class StopState {
+ public:
+  [[nodiscard]] bool IsStopped() const {
+    absl::MutexLock lock{&mutex_};
+    return stopped_;
+  }
+  void Stop() {
+    absl::MutexLock lock{&mutex_};
+    stopped_ = true;
+  }
+
+ private:
+  mutable absl::Mutex mutex_;
+  bool stopped_ = false;
```

Comment:
Yes, you understand correctly about the polling. But I am still sticking with the naming here. StopSource and StopToken are public as part of orbit_base and have the notions of requesting (RequestStop and IsStopRequested). This `orbit_base_internal::StopState` is an implementation detail that is independent from the "requesting action".  I am keeping the naming with `StopState`, `Stop()`, `IsStopped()` and `stopped_` instead of putting the verb request into all of these.